lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+
+  defstruct(
+    players: [],
+    npcs: [],
+    map: nil)
+
+  def join(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_join, player_entity})
+  end
+
+  def leave(entity, player_entity) do
```

Comment:
do we need this? or can we make this automatic, coordination based? <- should be the case since we add the players to the map's coord channel and they get removed automatically